hash.c

```diff
@@ -6842,6 +6842,129 @@ env_update(VALUE env, VALUE hash)
+ *  #has_value?:: Returns whether a given object is a value in +self+.
+ *  #include?, #has_key?, #member?, #key?:: Returns whether a given object is a key in +self+.
+ *  #length, #size:: Returns the count of entries.
+ *  #value?:: Returns whether a given object is a value in +self+.
+ *
+ *  ==== Methods for Comparing
+ *
+ *  {#<}[#method-i-3C]:: Returns whether +self+ is a proper subset of a given object.
+ *  {#<=}[#method-i-3C-3D]:: Returns whether +self+ is a subset of a given object.
+ *  {#==}[#method-i-3D-3D]:: Returns whether a given object is equal to +self+.
+ *  {#>}[#method-i-3E]:: Returns whether +self+ is a proper superset of a given object
+ *  {#>=}[#method-i-3E-3D]:: Returns whether +self+ is a proper superset of a given object.
+ *
+ *  ==== Methods for Fetching
+ *
+ *  #[]:: Returns the value associated with a given key.
+ *  #assoc:: Returns a 2-element array containing a given key and its value.
+ *  #dig:: Returns the object in nested objects that is specified
+ *         by a given key and additional arguments.
+ *  #fetch:: Returns the value for a given key.
+ *  #fetch_values:: Returns array containing the values associated with given keys.
+ *  #key:: Returns the key for the first-found entry with a given value.
+ *  #keys:: Returns an array containing all keys in +self+.
+ *  #rassoc:: Returns a 2-element array consisting of the key and value
+              of the first-found entry having a given value.
+ *  #values:: Returns an array containing all values in +self+/
+ *  #values_at:: Returns an array containing values for given keys.
+ *
+ *  ==== Methods for Assigning
+ *
+ *  #[]=, #store:: Associates a given key with a given value.
+ *  #merge:: Returns the hash formed by merging each given hash into a copy of +self+.
+ *  #merge!, #update:: Merges each given hash into +self+.
+ *  #replace:: Replaces the entire contents of +self+ with the contents of a givan hash.
+ *
+ *  ==== Methods for Deleting
+ *
+ *  These methods remove entries from +self+:
+ *
+ *  #clear:: Removes all entries from +self+.
+ *  #compact!:: Removes all +nil+-valued entries from +self+.
+ *  #delete:: Removes the entry for a given key.
+ *  #delete_if:: Removes entries selected by a given block.
+ *  #filter!, #select!:: Keep only those entries selected by a given block.
+ *  #keep_if:: Keep only those entries selected by a given block.
+ *  #reject!:: Removes entries selected by a given block.
+ *  #shift:: Removes and returns the first entry.
+ *  #slice:: Returns a hash containing the entries for given keys.
+ *
+ *  These methods return a copy of +self+ with some entries removed:
+ *
+ *  #compact:: Returns a copy of +self+ with all +nil+-valued entries removed.
+ *  #except:: Returns a copy of +self+ with entries removed for specified keys.
+ *  #filter, #select:: Returns a copy of +self+ with entries removed as specified by a given block.
+ *  #reject:: Returns a copy of +self+ with entries removed as specified by a given block.
```

Comment:
`#filter`/`#select` and `#reject` have the opposite behavior, you need to use different text for them.  You could change `removed` to `kept` (or `removed except`) for `#filter`/`#select`  to fix this.

Reply:
#filter/#select now similar to #filter!/#select! above:
 *  #filter, #select:: Returns a copy of +self+ with only those entries selected by a given block.